test/browser.test.tsx

```diff
@@ -25,7 +27,13 @@ beforeEach(() => {
     //   (`Component`). To locate the bad setState() call inside `Component`, follow the stack trace
     //   as described in https://reactjs.org/link/setstate-in-render"
     vi.spyOn(console, 'error').mockImplementation((format: string, ...args: any[]) => {
```

Comment:
I do like this pattern. 

What I've done in the past is maintained a list of allowed errors, and a second list of errors that should be printed to the console.